beacon-chain/sync/service.go

```diff
@@ -100,10 +103,33 @@ type Service struct {
 	blockNotifier        blockfeed.Notifier
 	blocksRateLimiter    *leakybucket.Collector
 	attestationNotifier  operation.Notifier
+	seenBlockLock        sync.RWMutex
+	seenBlockCache       *ristretto.Cache
+	seenAttestationLock  sync.RWMutex
+	seenAttestationCache *ristretto.Cache
 }
 
 // Start the regular sync service.
 func (r *Service) Start() {
+	bCache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: seenBlockSize, // Max size is seenBlockSize * 32byte.
+		MaxCost:     seenBlockSize / 10,
+		BufferItems: 64,
+	})
+	if err != nil {
+		panic(err)
+	}
+	aCache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: seenAttSize, // Max size is seenAttSize * 32byte.
+		MaxCost:     seenAttSize / 10,
```

Comment:
You are costing values as 1. What is the intended number of values to keep in the cache?

Reply:
I'm budgeting 100 for blocks and 1000 for attestations. With proper dynamic committee subnet, there's roughly 10x more attestations than block on per slot level coming in via the wire. I definitely over-budged a bit but these are cheap, just a 32byte key to bool